src/suricata.c

```diff
@@ -336,6 +336,7 @@ static void GlobalsDestroy(SCInstance *suri)
     }
     DetectEnginePruneFreeList();
 
+    // XXX Prefer to have de_ctx available here
```

Comment:
Not needed AFAICS. The app-layer doesn't use a detect engine, so pass NULL to any detect keyword free func that might it